common/Models/ExtensionAdaptiveCard.cs

```diff
@@ -20,11 +24,14 @@ public class ExtensionAdaptiveCard : IExtensionAdaptiveCard
 
     public string TemplateJson { get; private set; }
 
-    public ExtensionAdaptiveCard()
+    public ExtensionAdaptiveCard(AdaptiveElementParserRegistration? elementParserRegistration = null, AdaptiveActionParserRegistration? actionParserRegistration = null)
     {
         TemplateJson = new JsonObject().ToJsonString();
         DataJson = new JsonObject().ToJsonString();
         State = string.Empty;
+
+        _elementParserRegistration = elementParserRegistration != null ? elementParserRegistration : new AdaptiveElementParserRegistration();
+        _actionParserRegistration = actionParserRegistration != null ? actionParserRegistration : new AdaptiveActionParserRegistration();

```

Comment:
Total nit, but looking again I think it can be even shorter:
```suggestion
        _elementParserRegistration = elementParserRegistration ?? new AdaptiveElementParserRegistration();
        _actionParserRegistration = actionParserRegistration ?? new AdaptiveActionParserRegistration();
```